.github/workflows/ci.yml

```diff
@@ -0,0 +1,48 @@
+name: CI
+
+on:
+  push:
+    branches: [master]
+  pull_request:
```

Comment:
You could specify branches for this one, which would mean it would only apply to pull requests made against main / master. I think this is more standard practice (this is what the github templates suggest IME), but 🤷 .